### PR TITLE
Make "Generic exception while checking for reachability" log less noisy

### LINKED ISSUE #15109: cli.System.UriFormatException: Invalid URI: The hostname could not be parsed.
**Describe the bug**
Exception reported on mountain duck start.

Mountain duck appears to be trying to connect to a blank URL, for reasons unknown.

**To Reproduce**
1. Open mountain duck debug log
2. See exception at top of log.

**Expected behavior**
No exception.

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 4.14.3

**Log Files**

```
2023-08-29 10:39:56,046 [Thread-7] ERROR Ch.Cyberduck.Core.Diagnostics.TcpReachability - Generic exception while checking for reachability for Host{protocol=Profile{parent=davs, vendor=iterate GmbH, description=null, image=null}, region='null', port=443, hostname='', credentials=Credentials{user='', oauth='OAuthTokens{accessToken='', refreshToken='', expiryInMilliseconds=9223372036854775807}', token='', identity=null}, uuid='null', nickname='null', defaultpath='null', workdir=null, custom=null, labels=[]}
cli.System.UriFormatException: Invalid URI: The hostname could not be parsed.
        at cli.System.Uri.CreateThis(Unknown Source) ~[System.dll:?]
        at cli.System.Net.WebRequest.CreateHttp(Unknown Source) ~[System.dll:?]
        at cli.Ch.Cyberduck.Core.Diagnostics.TcpReachability.isReachable(Unknown Source) ~[Cyberduck.Core.Native.DLL:?]
        at java.lang.Throwable.getLocalizedMessage(map.xml:1230) ~[?:1.8.0]
        at org.apache.logging.log4j.core.impl.ThrowableProxy.<init>(ThrowableProxy.java:106) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.impl.ThrowableProxy.<init>(ThrowableProxy.java:98) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.impl.MutableLogEvent.getThrownProxy(MutableLogEvent.java:354) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.pattern.ExtendedThrowablePatternConverter.format(ExtendedThrowablePatternConverter.java:63) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.pattern.PatternFormatter.format(PatternFormatter.java:44) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.layout.PatternLayout$PatternFormatterPatternSerializer.toSerializable(PatternLayout.java:383) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.layout.PatternLayout.toText(PatternLayout.java:240) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.layout.PatternLayout.encode(PatternLayout.java:225) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.layout.PatternLayout.encode(PatternLayout.java:59) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.directEncodeEvent(AbstractOutputStreamAppender.java:216) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.tryAppend(AbstractOutputStreamAppender.java:208) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.append(AbstractOutputStreamAppender.java:199) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.appender.RollingFileAppender.append(RollingFileAppender.java:302) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:161) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:136) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:125) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:90) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:682) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:643) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:626) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:560) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.config.AwaitCompletionReliabilityStrategy.log(AwaitCompletionReliabilityStrategy.java:82) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.core.Logger.log(Logger.java:163) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.spi.AbstractLogger.tryLogMessage(AbstractLogger.java:2168) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.spi.AbstractLogger.logMessageTrackRecursion(AbstractLogger.java:2122) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.spi.AbstractLogger.logMessageSafely(AbstractLogger.java:2105) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.spi.AbstractLogger.logMessage(AbstractLogger.java:1981) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.spi.AbstractLogger.logIfEnabled(AbstractLogger.java:1948) [Cyberduck.Core.DLL:2.20.0]
        at org.apache.logging.log4j.spi.AbstractLogger.error(AbstractLogger.java:719) [Cyberduck.Core.DLL:2.20.0]
        at cli.Ch.Cyberduck.Core.Diagnostics.TcpReachability.isReachable(Unknown Source) [Cyberduck.Core.Native.DLL:?]
```

### PR DESCRIPTION
**Pull request by AliveDevil**
Resolves #15109

### CODE REVIEW

File: core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs [hunk 1]
```diff
@@ -81,31 +80,34 @@ public bool isReachable(Host h)
                             // not an exception?
                             case WebExceptionStatus.Success:
                                 return true;
-                        }
 
-                        return false;
+                            default:
+                                if (Log.isDebugEnabled())
+                                {
+                                    Log.debug($"WebException thrown with status {e.Status} for {h}");
```
Comment (AliveDevil) [inline core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs:87]: We don't need to log handled WebExceptionStatus codes.

File: core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs [hunk 2]
```diff
@@ -81,31 +80,34 @@ public bool isReachable(Host h)
...
-                default:
-                    try
-                    {
-                        if (Log.isDebugEnabled())
-                        {
-                            Log.debug($"Try TCP connection to {h.getHostname()}:{h.getPort()}");
-                        }
 
-                        TcpClient c = new TcpClient(h.getHostname(), h.getPort());
-                        c.Close();
```
Comment (AliveDevil) [inline core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs:102]: TcpClient.Dispose() automatically calls close, thus `using()` is sufficient.

File: core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs [hunk 3]
```diff
@@ -39,12 +39,16 @@ static TcpReachability()
 
         public bool isReachable(Host h)
         {
-            switch ((Scheme.__Enum)h.getProtocol().getScheme().ordinal())
+            var scheme = h.getProtocol().getScheme();
+            if (Scheme.file.equals(scheme))
```
Comment (AliveDevil) [inline core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs:43]: This doesn't need to be in a try-block, reducing complexity later.

File: core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs [hunk 4]
```diff
@@ -81,31 +80,34 @@ public bool isReachable(Host h)
...
+                    using (new TcpClient(h.getHostname(), h.getPort()))
                     {
-                        return false;
+                        return true;
                     }
+                }
+            }
+            catch (Exception e)
+            {
+                Log.warn($"Reachability check for {h} failed {e.Message}");
```
Comment (AliveDevil) [inline core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs:109]: One large catch around all of the networking stuff.
No need to have two catch blocks with exactly the same content (or rather: no content. Up until now reachability results for Tcp-tests didn't end up in the log)

File: core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs (hunk 3 above)
Comment (AliveDevil) [reply core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs:43]: Should we fast-exit for Hostname equals `""`?

Comment (dkocher) [reply core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs:43]: On macOS, no reachability check is made if the input is blank [^1]. 

[^1]: https://github.com/iterate-ch/cyberduck/blob/da4384c42831e87569bd3bc9ea1835c3399612d0/osx/src/main/java/ch/cyberduck/ui/cocoa/controller/BookmarkController.java#L287

Comment (AliveDevil) [reply core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs:43]: Ah I see. Applying to Mountain Duck as well.

**Review by dkocher — APPROVED** [review]

**Pull request merged**